misc/toml-parser/src/main/java/io/ballerina/toml/internal/parser/TomlLexer.java

```diff
@@ -261,27 +285,74 @@ private STToken readLiteralStringToken() {
         }
 
         char nextChar = this.reader.peek();
-        char secondNextChar = this.reader.peek(1);
-        char thirdNextChar = this.reader.peek(2);
         if (nextChar == LexerTerminals.SINGLE_QUOTE) {
             endMode();
-            if (secondNextChar == LexerTerminals.SINGLE_QUOTE && thirdNextChar == LexerTerminals.SINGLE_QUOTE) {
-                reader.advance(3);
-                return getSyntaxToken(SyntaxKind.TRIPLE_SINGLE_QUOTE_TOKEN);
-            }
             reader.advance();
             return getSyntaxToken(SyntaxKind.SINGLE_QUOTE_TOKEN);
-        } else {
-            while (!reader.isEOF()) {
-                nextChar = this.reader.peek();
-                if (nextChar != LexerTerminals.SINGLE_QUOTE) {
+        }
+        while (!reader.isEOF()) {
+            nextChar = this.reader.peek();
+            switch (nextChar) {
+                case LexerTerminals.SINGLE_QUOTE:
+                case LexerTerminals.CARRIAGE_RETURN:
+                case LexerTerminals.NEWLINE:
+                    break;
+                default:
                     reader.advance();
                     continue;
-                }
-                break;
             }
+            break;
+        }
+        return getUnquotedKey();
+    }
+
+    private STToken readMultilineStringToken() {
+        reader.mark();
+        if (reader.isEOF()) {
+            return getSyntaxToken(SyntaxKind.EOF_TOKEN);
         }
 
+        char nextChar = this.reader.peek();
+        char secondNextChar = this.reader.peek(1);
+        char thirdNextChar = this.reader.peek(2);
+        if (nextChar == LexerTerminals.DOUBLE_QUOTE && secondNextChar == LexerTerminals.DOUBLE_QUOTE && thirdNextChar
+                == LexerTerminals.DOUBLE_QUOTE) {
+            endMode();
+            reader.advance(3);
+            return getSyntaxToken(SyntaxKind.TRIPLE_DOUBLE_QUOTE_TOKEN);
+        }
+        while (!reader.isEOF()) {
+            nextChar = this.reader.peek();
+            if (nextChar == LexerTerminals.DOUBLE_QUOTE && this.reader.peek(1) == LexerTerminals.DOUBLE_QUOTE &&
```

Comment:
since `nextChar == LexerTerminals.DOUBLE_QUOTE && this.reader.peek(1) == LexerTerminals.DOUBLE_QUOTE && this.reader.peek(2) == LexerTerminals.DOUBLE_QUOTE` is used at L#318 as well, we can take it to a common method